### DISCUSSION TITLE
将新版 opus “图文”页面重定向回旧版动态和专栏页面

### DISCUSSION THREAD

**Discussion by GalaxySnail** [the1812/Bilibili-Evolved#4596] [功能建议 / Ideas]
### 在动态页打开专栏会前往 opus 页面

从数个月前的某个时间开始，在动态页（包括 t.bilibili.com 和up主个人主页的 dynamic 动态页）中打开专栏不会直接打开 `www.bilibili.com/read/cvXXX` 的专栏页面，而是打开“相簿”页面的 URL，形如 `www.bilibili.com/opus/XXXXX`。之后普通专栏会自动重定向到 `/read/cvXXX` 页面，而视频笔记（本质上仍是有cv号的专栏）则会停留在 `/opus/XXXXX` 页面，需要手动点击“返回旧版”才能回到专栏的 `/read/cvXXX` 页面。例如：

![](https://github.com/the1812/Bilibili-Evolved/assets/53637421/ce72ee11-71fb-40da-95c8-ace7384e578b)

- opus 链接：https://www.bilibili.com/opus/883845303941201956
- 专栏链接：https://www.bilibili.com/read/cv29130272/

在那时，仍然可以通过在up主个人主页的投稿页面，选择“专栏”分类从而直接进入 `/read/cvXXX` 页面。

### “相簿”和“专栏”被合并为“图文”

而最近，B站开始灰度测试合并up主个人主页投稿页面的“相簿”（带图片的动态）和“专栏”，合二为一成“图文”类别。旧版页面如下（以 https://space.bilibili.com/2039207951/article 为例）：

![](https://github.com/the1812/Bilibili-Evolved/assets/53637421/6ee6e1b4-2a07-400e-902f-669db5aae56b)

新版页面如下：

![](https://github.com/the1812/Bilibili-Evolved/assets/53637421/6ed2175c-4b85-4df4-8a64-0393f181b283)

这导致了两个明显的结果：

- 无法再快速检索一名up主投稿的所有专栏（被带图动态污染）
- 从新的图文页面点开专栏也会进入 `/opus/XXXXX` 页面，而不是直接进入 `/read/cvXXX` 页面

## 建议

此功能请求主要是针对第二个结果。希望可以在 `www.bilibili.com/opus/XXXXX` 页面自动执行重定向，对于动态则重定向至 `t.bilibili.com/XXXXX`，对于专栏则重定向至 `www.bilibili.com/read/cvXXX`。

## 相簿专栏合并的复现步骤

目前此合并似乎仍处于灰度状态，开无痕浏览有时会合并有时则不会。我经过测试发现这应该与 `buvid3` cookie 有关，首先打开up主投稿页面的 article 类别，删除此 cookie 后刷新两次页面，会随机处于已合并或未合并的状态，此后不删除 cookie 再刷新就会一直保持在已合并或未合并状态。

**Comment by the1812** [comment]
看了下空间里那个 API (`https://api.bilibili.com/x/polymer/web-dynamic/v1/opus/feed/space`), 似乎并没有给出两种类型的区分字段

**Comment by qtqz** [comment]
第二件事，现在好像强制新版了

**Comment by Alice0000000** [comment]
支持
